testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeCreateOrReplaceTableAsSelectCompatibility.java

```diff
@@ -0,0 +1,195 @@
+        onTrino().executeQuery("CREATE TABLE delta.default." + tableName + " (ts VARCHAR) " +
+                "with (location = 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "', checkpoint_interval = 10)");
+        try {
+            ImmutableList.Builder<Row> expected = ImmutableList.builder();
+            expected.addAll(performInsert(onDelta(), tableName, 9));
+
+            onDelta().executeQuery("CREATE OR REPLACE TABLE " + tableName + " USING DELTA AS SELECT CAST(ts AS TIMESTAMP) FROM " + tableName);
+            assertLastEntryIsCheckpointed(s3, bucketName, tableName);
+
+            onDelta().executeQuery("INSERT INTO " + tableName + " VALUES \"1960-01-01 01:02:03\", \"1961-01-01 01:02:03\", \"1962-01-01 01:02:03\"");
+            expected.add(row("1960-01-01T01:02:03.000Z"), row("1961-01-01T01:02:03.000Z"), row("1962-01-01T01:02:03.000Z"));
+            assertTransactionLogVersion(s3, bucketName, tableName, 11);
+
+            assertThat(onTrino().executeQuery("SELECT to_iso8601(ts) FROM delta.default." + tableName)).containsOnly(expected.build());
+        }
+        finally {
+            dropDeltaTableWithRetry(tableName);
+        }
+    }
+
+    @Test(groups = {DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    public void testCreateOrReplaceTableOnAppendOnlyTableFails()
+    {
+        String tableName = "test_replace_on_append_only_table_fails_" + randomNameSuffix();
+        onDelta().executeQuery("" +
+                "CREATE TABLE default." + tableName +
+                "         (a INT, b INT)" +
+                "         USING delta " +
+                "         LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "' " +
+                "         TBLPROPERTIES ('delta.appendOnly' = true)");
+        try {
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES (1,11), (2, 12)");
+            assertQueryFailure(() -> onDelta().executeQuery("" +
+                    "CREATE OR REPLACE TABLE default." + tableName +
+                    "         (a INT,c INT) " +
+                    "         USING delta " +
+                    "         LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "' " +
+                    "         TBLPROPERTIES ('delta.appendOnly' = true)"))
+                    .hasMessageContaining("This table is configured to only allow appends");
+            assertQueryFailure(() -> onDelta().executeQuery("" +
+                    "CREATE OR REPLACE TABLE default." + tableName +
+                    "         (a INT,c INT) " +
+                    "         USING delta " +
+                    "         LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "' " +
+                    "         TBLPROPERTIES ('delta.appendOnly' = false)"))
+                    .hasMessageContaining("This table is configured to only allow appends");
```

Comment:
Those statemetns feel duplicated. --intentional?

Reply:
The value of `delta.appendOnly` differ.